theme/scripts/components/Component.ts

```diff
@@ -4,18 +4,18 @@ export interface NamedComponent {
 	componentName: string
 }
 
-export type ComponentConstructor<D> = NamedComponent &
-	(new (element: HTMLElement, data: D) => Component<D>)
+export type ComponentConstructor<D, E extends HTMLElement = HTMLElement> = NamedComponent &
+	(new (element: E, data: D) => Component<D, E>)
 
 export class ComponentInitializationError extends Error {}
 
-export default class Component<D> {
-	protected readonly el: HTMLElement
+export default class Component<D, E extends HTMLElement = HTMLElement> {
+	protected readonly el: E
 	protected readonly data: D
 
 	getListeners = (): EventListeners => []
 
-	constructor(element: HTMLElement, data: D) {
+	constructor(element: E, data: D) {
```

Comment:
I'm getting lost in the naming convention for `E`,  `C`, `D`, `F` ... what these stand for and can we name them fully?